cmd/coder/users.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/pflag"
+
+	"go.coder.com/cli"
+	"go.coder.com/flog"
+)
+
+type usersCmd struct {
+}
+
+func (cmd usersCmd) Spec() cli.CommandSpec {
+	return cli.CommandSpec{
+		Name:  "users",
+		Usage: "[subcommand] <flags>",
+		Desc:  "used to interact with user accounts",
+	}
+}
+
+func (cmd usersCmd) Run(fl *pflag.FlagSet) {
+	exitUsage(fl)
+}
+
+func (cmd *usersCmd) Subcommands() []cli.Command {
+	return []cli.Command{
+		&listCmd{},
+	}
+}
+
+type listCmd struct {
+	outputFmt string
+}
+
+func tabDelimited(data interface{}) string {
+	v := reflect.ValueOf(data)
+	s := &strings.Builder{}
+	for i := 0; i < v.NumField(); i++ {
+		s.WriteString(fmt.Sprintf("%s\t", v.Field(i).Interface()))
+	}
+	return s.String()
+}
+
+func (cmd *listCmd) Run(fl *pflag.FlagSet) {
+	entClient := requireAuth()
+
+	users, err := entClient.Users()
+	if err != nil {
+		flog.Fatal("failed to get users: %v", err)
+	}
+
+	switch cmd.outputFmt {
+	case "human":
+		w := tabwriter.NewWriter(os.Stdout, 0, 0, 3, ' ', 0)
+		for _, u := range users {
+			_, err = fmt.Fprintln(w, tabDelimited(u))
+			if err != nil {
+				flog.Fatal("failed to write: %v", err)
+			}
+		}
+		err = w.Flush()
+		if err != nil {
+			flog.Fatal("failed to flush writer: %v", err)
+		}
+	case "json":
+		err = json.NewEncoder(os.Stdout).Encode(users)
+		if err != nil {
+			flog.Fatal("failed to encode users to json: %v", err)
+		}
+	default:
+		flog.Fatal("unsupported output type %q.\nusage: --output (json | human)", cmd.outputFmt)
+	}
+
+}
+
+func (cmd *listCmd) RegisterFlags(fl *pflag.FlagSet) {
+	fl.StringVarP(&cmd.outputFmt, "output", "o", "human", "specify the output format")
+}
+
+func (cmd *listCmd) Spec() cli.CommandSpec {
+	return cli.CommandSpec{
+		Name:  "ls",
+		Usage: "<flags>",
+		Desc:  "list all users",
```

Comment:
```suggestion
		Desc:  "list all users that the signed-in account has admin access to sitewide",
```

Reply:
All users have perms to list other users, not just admin.

Reply:
It'd be nice to make this clear either way and use the help to tell us more than what the command already does e.g `users ls` says `lists users` but it's not clear what users

Reply:
list _all_ users